src/query/builder.js

```diff
@@ -751,11 +755,23 @@ assign(Builder.prototype, {
     let amt = parseInt(amount, 10);
     if (isNaN(amt)) amt = 1;
     this._method = 'counter';
-    this._single.counter = {
-      column,
-      amount: amt,
-      symbol: (symbol || '+')
-    };
+    let counter = this._single.counter || {};
+
+    if(!counter[column]) {
+      counter[column] = {
+        amount: amt,
+        symbol: (symbol || '+')
+      };
+    } else {
+      //Counter for column already exists. Add/subtract the current column amount by the new amount
+      if(symbol === '-') {
+        counter[column].amount -= amount;
```

Comment:
@elhigu In pratical terms that would be pretty annoying, imo. For example if you're doing an application where you need to loop to calculate a new value, and each iteration can contain either an addition or a subtraction, you shouldn't have to store the identifier (+/-) in a seperate variable to know whether to call increment/decrement, nor should you have to store the final value that is to be updated in a seperate variable.

It would be **a lot** easier and far more convenient while in the loop to simply call queryBuilder.increment/decrement with the value of the current iteration, which eventually results in the correct query when executed. That's my 2-cents on this.

In any case I'm not sure why there's a need to review something that I've already closed, this issue needs more thought around the `insert` function as the current implementation won't allow an easy fix.